src/index/CompressedRelation.cpp

```diff
@@ -87,9 +89,9 @@ IdTable CompressedRelationReader::scan(
         // The `decompressLambda` can now run in parallel
 #pragma omp task
         {
-          if (!timer || !timer->wlock()->hasTimedOut()) {
-            decompressLambda();
-          };
+          // TODO verify correct behaviour
```

Comment:
Is throwing fine with pragma omp?

Reply:
It is not and will immediately crash your program (basically you throw inside a plain thread with no catching.
You should add a functionality to youyr handle that just checks with a simple bool whether it was canceled (only the relaxed read basically).

Reply:
As long as we are in the `pragma omp` world, we have a problem. So no, this doesn't work, you have to gracefully ask for the cancellation without throwing and we need this api to make this work in cases where throwing is not an option (we will have those for a while).